File: app/src/main/java/com/wilinz/yuetingmusic/data/model/RecentSong.java
package com.wilinz.yuetingmusic.data.model;

import org.litepal.annotation.Column;
import org.litepal.crud.LitePalSupport;

public class RecentSong extends LitePalSupport {
    public int id;
    public User user;
    public Song song;
    public long song_id;
    public long timestamp;
    @Column(nullable = false, unique = true, index = true)
    public String uniqueId;
}

File: app/src/main/java/com/wilinz/yuetingmusic/data/model/RecentSong.kt
package com.wilinz.yuetingmusic.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import org.litepal.annotation.Column
import org.litepal.crud.LitePalSupport

@Parcelize
data class RecentSong(
    var id: Int = 0,
    var user: User? = null,
    var song: Song? = null,
    var song_id: Long = 0,
    var timestamp: Long = 0,
    @Column(nullable = false, unique = true, index = true)
    var uniqueId: String? = null,
) : Parcelable, LitePalSupport()
